src/NumFieldOrd/NfOrd/ResidueField.jl

```diff
@@ -210,3 +210,45 @@ function ResidueFieldSmallDegree1(O::NfOrd, P::NfOrdIdl)
     return _residue_field_generic(O, P, Val{true}, Val{true})
   end
 end
+
+@doc Markdown.doc"""
+    relative_residue_field(O::NfRelOrd, P::NfRelOrdIdl) -> RelFinField, Map
+
+Given a maximal order `O` in a relative number field $E/K$ and a prime ideal 
+`P` of `O`, return the residue field $O/P$ seen as an extension of the (relative) 
+residue field of a maximal order in `K` at $minimum(P)$.
+
+Note that if `K` is a relative number field, the latter will also be seen as a 
+relative residue field.
+
+The map from `O` to the residue field is stored in `E` under the attribute 
+`:rel_residue_field_map`.
+"""
+function relative_residue_field(O::NfRelOrd{S, T, U}, P::NfRelOrdIdl{S, T, U}) where {S, T, U}
+  @req ismaximal(O) "O must be maximal"
+  @req order(P) === O "P must be an ideal of O"
+  E = nf(O)
+  K = base_field(E)
+  projK = get_attribute(K, :rel_residue_field_map)
```

Comment:
I am not sure I understand why we want to store it on the base field of the number field. What about storing it on `P`? I

Reply:
For now I made this choice because I wanted to use it for the completions of tower of extensions. So the idea would be to get the relative residue field of the top field, and get access to the other maps  by checking the successive base fields. But I am not close to any change/improvement :)

Reply:
Yes, but the result here will just be wrong if `relative_residue_field(O, P)` is called for different `P`s? Or am I missing something.

Reply:
Ah yes indeed, I haven't thought about this... that's a very good point. So yes, it should be more reasonable to store it on the ideal.